server/lib/cron-scheduler/save-document.ts

```diff
@@ -40,13 +40,60 @@ export class SaveDocument {
       throw error;
     }
   }
+  private async checkIndexPatternAndCreateIfNotExists(index) {
+    const KIBANA_INDEX = this.context.server.config.kibana.index;
+    log(this.logPath, `Internal index of kibana: ${KIBANA_INDEX}`, 'debug');
+    const result = await this.esClientInternalUser.search({
+      index: KIBANA_INDEX,
+      type: '_doc',
+      body: {
+        query: {
+          match: {
+            _id: `index-pattern:${index}-*`
+          }
+        }
+      }
+    });
+    if (result.body.hits.total.value === 0) {
+      await this.createIndexPattern(KIBANA_INDEX, index);
+    }
+  }
+
+  private async createIndexPattern(KIBANA_INDEX: any, index: any) {
+    try {
+      const response = await this.esClientInternalUser.create({
+        index: KIBANA_INDEX,
+        type: '_doc',
+        'id': `index-pattern:${index}-*`,
+        body: {
+          type: 'index-pattern',
+          'index-pattern': {
+            title: `${index}-*`,
+            timeFieldName: 'timestamp',
+          }
+        }
+      });
+      log(
+        this.logPath,
+        `The indexPattern no exist, response of createIndexPattern: ${JSON.stringify(response)}`,
+        'debug'
+      );
+    } catch (error) {
+      this.checkDuplicateIndexPatterError(error);
+    }
+  }
```

Comment:
This logic creates an index pattern in the backend. We create it in the frontend. You could remove these methods too.